src/foundry/foundry.js/application.d.ts

```diff
@@ -75,15 +75,37 @@ declare class Application<P extends Application.Options = Application.Options> {
 
```

Comment:
Foundry now has a type definition for the options (which seems to be incomplete but includes options we don't have in our definition):
```js
/**
 * @typedef {Object} ApplicationOptions
 * @property {boolean} [closeOnSubmit=true]     Whether to automatically close the application when it's contained
 *                                              form is submitted.
 * @property {boolean} [submitOnChange=false]   Whether to automatically submit the contained HTML form when an input
 *                                              or select element is changed.
 * @property {boolean} [submitOnClose=false]    Whether to automatically submit the contained HTML form when the
 *                                              application window is manually closed.
 * @property {boolean} [editable=true]          Whether the application form is editable - if true, it's fields will
 *                                              be unlocked and the form can be submitted. If false, all form fields
 *                                              will be disabled and the form cannot be submitted.
 */
```

Reply:
I don't think that these options should go to the application. If you look at the parameter description, these are the options of the `FormApplication` (and there correctly described). They are also never mentioned in the `Application` class (but in the `FormApplication` class).

Reply:
OK, I still think we should create the type. Maybe create it in `formApplication.d.ts` and have `FormApplication.Options` extend bot `Application.Options` and `ApplicationOptions`?